src/main/webapp/scripts/my-trips.js

```diff
@@ -196,23 +196,99 @@ async function findHotel() {
     modalContent.innerText = "No hotels nearby. Sorry.";
   } else {
-    json.results = json.results.slice(0, 4);
-    modalContent.innerHTML = json.results
+    const hotelsMapElem = document.getElementById("hotels-map");
+    json = json.slice(0, 10);
+    const hotelMap = new google.maps.Map(
+      document.getElementById("hotels-map"),
+      {
+        center: centerPoint,
+        zoom: 13,
+      }
+    );
+    // Add existing locations to the map
+    markers.forEach((obj) => {
+      const { position } = obj;
+      const { lat, lng } = position;
+      const marker = new google.maps.Marker({
+        position: { lat: lat(), lng: lng() },
+        map: hotelMap,
+        title: obj.title,
+      });
+    });
+    // Add distance_center and photo_url fields to each object in
+    // json.results
+    json = json.map(async (obj) => {
+      const { geometry } = obj;
+      const { location } = geometry;
+      const marker = new google.maps.Marker({
+        position: location,
+        map: hotelMap,
+        title: obj.name,
+        label: {
+          fontFamily: "Material Icons",
+          text: "hotel",
+        },
+      });
+      const infoWindow = new google.maps.InfoWindow({
+        content: `<h5 class="infowindow-text">${obj.name}</h5>`,
+      });
+      marker.addListener("click", () => infoWindow.open(hotelMap, marker));
+      obj.distance_center = distanceBetween(location, centerPoint);
+      const photoRef = obj.photos[0]
+        ? obj.photos[0].photo_reference
+        : undefined;
+      if (photoRef) {
+        const photoResponse = await fetch(
+          `https://cors-anywhere.herokuapp.com/https://maps.googleapis.com/maps/api/place/photo?maxwidth=500&photoreference=${photoRef} &key=${GOOGLE_API_KEY}`
+        );
+        const blob = await photoResponse.blob();
+        const photoUrl = await URL.createObjectURL(blob);
+        obj.photo_url = photoUrl;
+      } else {
+        obj.photo_url = undefined;
+      }
+      return await obj;
+    });
+    json = await Promise.all(json);
+    json.sort((a, b) => a.distance_center - b.distance_center);
+    hotelsMapElem.style.width = "100%";
+    hotelsMapElem.style.height = "400px";
+    hotelsMapElem.style.marginBottom = "2em";
+    modalContent.innerHTML = json
       .map(
-        ({ name, formatted_address, rating, place_id }) => `
-          <div class="col s6">
-            <div class="card white">
-              <div class="card-content black-text">
-                <span class="card-title"><strong>${name}</strong> | Rating: ${rating}</span>
-                <p>${formatted_address}</p>
+        ({ name, formatted_address, rating, place_id, photo_url }) =>
+          `
+          <div class="row">
+            <div class="col s12 m6">
+              <div class="card large white">` +
+          (photo_url
+            ? `
+                <div class="card-image">
+                  <img src="${photo_url}" alt="photo of ${name} from Google" loading="lazy" />
```

Comment:
What does lazy loading do?

Reply:
Lazy Loading is basically when instead of initializing an object say at launch or on load, you only wait to initialize that object until after it is needed or called upon. I believe in this case is being used to load the photo only when that section is rolled to possible to limit the initial spike that page would receive. Correct me if I'm wrong though Peter!

Reply:
> What does lazy loading do?

Lazy Loading is basically when instead of initializing an object say at launch or on load, you only wait to initialize that object until after it is needed or called upon. I believe in this case is being used to load the photo only when that section is rolled to possible to limit the initial spike that page would receive. Correct me if I'm wrong though Peter!

Reply:
Ah I see, thanks Josh!